website/www/site/content/en/contribute/runner-guide.md

```diff
@@ -89,6 +117,13 @@ But if your data is arriving as a stream, then you will want to terminate a
 bundle in order to achieve appropriate latency, so bundles may be just a few
 elements.
 
+A bundle is the unit of commitment in Beam. If an error is encountered while
+processing a bundle, all the prior outputs of that bundle (including any
+modifications to state or timers) must be discarded and the entire bundle
+retried.  Upon successful completion of a bundle, its outputs, together with
+any state/timer modifications and watermark updates, must be committed
+atomically.
+
 #### The DoFn Lifecycle
 
 While each language's SDK is free to make different decisions, the Python and
```

Comment:
I mostly left this here because I wasn't sure it was documented elsewhere, but it turns out it is so I'm just pointing to a reference and leaving this more portable-centric. 

(Agree this could use a much more invasive portable-first refactoring, but given limited time I wanted to focus on getting the basic points out first.)